### PR TITLE
Enhance the designer to add the PDF emitter options like configuration at advanced property register (#1844)

### PR DESCRIPTION
**Pull request by speckyspooky**

### CODE REVIEW

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 1]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 *
+	 * @param iccProfileExternal external color profile
+	 * @throws SemanticException
+	 */
+	public void setPdfIccColorProfileExternal(String iccProfileExternal) throws SemanticException {
+		setStringProperty(PDF_ICC_PROFILE_EXTERNAL, iccProfileExternal);
+	}
+
+	/**
+	 * Get the configuration to prepend document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1761]:
```suggestion
	 * Get the configuration to prepend document(s) to the PDF document
```

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 2]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 *
+	 * @return the configuration to prepend document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1763]:
```suggestion
	 * @return the configuration to prepend document(s) to the PDF document
```

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 3]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 */
+	public String getPdfDocumentsPrepend() {
+		return getStringProperty(PDF_DOCUMENTS_PREPEND);
+	}
+
+	/**
+	 * Set the configuration to prepend document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1770]:
```suggestion
	 * Set the configuration to prepend document(s) to the PDF document
```

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 4]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 *
+	 * @param prependDocuments document(s) to prepend
+	 * @throws SemanticException
+	 */
+	public void setPdfDocumentsPrepend(String prependDocuments) throws SemanticException {
+		setStringProperty(PDF_DOCUMENTS_PREPEND, prependDocuments);
+	}
+
+	/**
+	 * Get the configuration to append document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1780]:
```suggestion
	 * Get the configuration to append document(s) to the PDF document
```

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 5]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 *
+	 * @return the configuration to append document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1782]:
```suggestion
	 * @return the configuration to append document(s) to the PDF document
```

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java [hunk 6]
```diff
@@ -1680,4 +1680,176 @@ public String getExcelTemplateFile() {
...
+	 */
+	public String getPdfDocumentsAppend() {
+		return getStringProperty(PDF_DOCUMENTS_APPEND);
+	}
+
+	/**
+	 * Set the configuration to append document(s) at the PDF document
```
Comment (merks) [inline model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java:1789]:
```suggestion
	 * Set the configuration to append document(s) to the PDF document
```

File: engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/PDFPage.java [hunk 7]
```diff
@@ -470,9 +470,10 @@ private void drawText(String text, float textX, float textY, FontInfo fontInfo,
 			if (this.pageDevice.isPdfAFormat() && fontInfo.getBaseFont() != null
 					&& !fontInfo.getBaseFont().isEmbedded()) {
 				try {
+					// PDF/A fallback font must be entered full qualified with path and file name
```
Comment (merks) [inline engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/PDFPage.java:473]:
```suggestion
					// PDF/A fallback font must be entered fullly qualified with path and file name
```

**Review by merks — APPROVED** [review]
Just some minor grammar changes.

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 1 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done.

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 3 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 4 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 5 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 6 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done

File: engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/PDFPage.java (hunk 7 above)
Comment (speckyspooky) [reply engine/org.eclipse.birt.report.engine.emitter.pdf/src/org/eclipse/birt/report/engine/emitter/pdf/PDFPage.java]: done

File: model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java (hunk 2 above)
Comment (speckyspooky) [reply model/org.eclipse.birt.report.model/src/org/eclipse/birt/report/model/api/ReportDesignHandleImpl.java]: done

**Pull request merged**